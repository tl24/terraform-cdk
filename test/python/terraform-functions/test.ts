/**
 *
 * @group python
 */

import { TestDriver } from "../../test-helper";

describe("python terraform functions test synth", () => {
  let driver: TestDriver;

  beforeAll(async () => {
    driver = new TestDriver(__dirname);
    await driver.setupPythonProject();
  });

  test("synth generates JSON", async () => {
    await driver.synth();
    expect(driver.synthesizedStack("python-simple")).toMatchSnapshot();
  });
});
