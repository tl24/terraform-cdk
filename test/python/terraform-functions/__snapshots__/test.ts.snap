// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`python full integration test synth synth generates JSON 1`] = `
"{
  \\"//\\": {
    \\"metadata\\": {
      \\"version\\": \\"stubbed\\",
      \\"stackName\\": \\"python-simple\\",
      \\"backend\\": \\"local\\",
      \\"overrides\\": {
        \\"stack\\": [
          \\"terraform\\"
        ],
        \\"aws_sns_topic\\": [
          \\"display_name\\"
        ]
      }
    }
  },
  \\"terraform\\": {
    \\"required_providers\\": {
      \\"aws\\": {
        \\"version\\": \\"~> 2.0\\",
        \\"source\\": \\"aws\\"
      }
    },
    \\"backend\\": {
      \\"remote\\": {
        \\"organization\\": \\"test\\",
        \\"workspaces\\": {
          \\"name\\": \\"test\\"
        }
      }
    }
  },
  \\"provider\\": {
    \\"aws\\": [
      {
        \\"region\\": \\"eu-central-1\\"
      }
    ]
  },
  \\"module\\": {
    \\"CustomVpc\\": {
      \\"name\\": \\"custom-vpc\\",
      \\"source\\": \\"terraform-aws-modules/vpc/aws\\",
      \\"version\\": \\"2.77.0\\",
      \\"//\\": {
        \\"metadata\\": {
          \\"path\\": \\"python-simple/CustomVpc\\",
          \\"uniqueId\\": \\"CustomVpc\\"
        }
      }
    }
  },
  \\"resource\\": {
    \\"aws_sns_topic\\": {
      \\"Topic\\": {
        \\"display_name\\": \\"my-first-sns-topic\\",
        \\"//\\": {
          \\"metadata\\": {
            \\"path\\": \\"python-simple/Topic\\",
            \\"uniqueId\\": \\"Topic\\"
          }
        }
      }
    }
  }
}"
`;
